mpsl/include/mpsl_clock.h

```diff
@@ -0,0 +1,125 @@
+    MPSL_CLOCK_LF_SRC_XTAL  = 1, /**< LFCLK crystal oscillator. */
+    MPSL_CLOCK_LF_SRC_SYNTH = 2, /**< LFCLK Synthesized from HFCLK. */
+};
+
+/** @brief Recommended RC clock calibration timer interval. */
+#define MPSL_RECOMMENDED_RC_CTIV        16
+
+/** @brief Recommended RC clock calibration timer interval for temperature changes. */
+#define MPSL_RECOMMENDED_RC_TEMP_CTIV   2
+
+/** @brief Type representing LFCLK oscillator source. */
+typedef struct
+{
+    uint8_t source;         /**< LF oscillator clock source, see @ref MPSL_CLOCK_LF_SRC. */
+    uint8_t rc_ctiv;        /**< Only for ::MPSL_CLOCK_LF_SRC_RC.
+                                 Calibration timer interval in 1/4 second units.
+                                 @note To avoid excessive clock drift, 0.5 degrees Celsius is the
+                                       maximum temperature change allowed in one calibration timer
+                                       interval. The interval should be selected to ensure this.
+                                 @note Must be 0 if source is not ::MPSL_CLOCK_LF_SRC_RC. */
+    uint8_t rc_temp_ctiv;   /**< Only for ::MPSL_CLOCK_LF_SRC_RC: How often (in number of calibration
+                                 intervals) the RC oscillator shall be calibrated if the temperature
+                                 hasn't changed.
+                                      0: Always calibrate even if the temperature hasn't changed.
+                                      1: Only calibrate if the temperature has changed.
+                                      2-33: Check the temperature and only calibrate if it has changed,
+                                            however calibration will take place every rc_temp_ctiv
+                                            intervals in any case.
+
+                                 @note Must be 0 if source is not ::MPSL_CLOCK_LF_SRC_RC.
+
+                                 @note The application must ensure calibration at least once every
+                                       8 seconds to ensure +/-500 ppm clock stability.
+                                       The recommended configuration for ::MPSL_CLOCK_LF_SRC_RC
+                                       is given by @ref MPSL_RECOMMENDED_RC_CTIV
+                                       and @ref MPSL_RECOMMENDED_RC_TEMP_CTIV.
+                                       This sets the calibration interval to 4 seconds and guarantees
+                                       clock calibration every second calibration interval. That is,
+                                       the clock will be calibrated every 8 seconds.
+                                       If the temperature changes more that 0.5 every 4 seconds, the
+                                       clock will be calibrated every 4 seconds. See the
+                                       Product Specification for more information. */
+} mpsl_clock_lf_cfg_t;
```

Comment:
It could be worth it to consider having HF and LF clock APIs consistent. we use `mpsl_clock_lf` and `mpsl_clock_hfclk`. It looks a bit strange. @sofu1 what is your opinion on this?

Reply:
Yes I agree. It should be consistent. I think we should align it with the HF clock API and use mpsl_clock_lfclk_.. Is it ok to do this for the next release though?

Reply:
Decided that it is ok to postpone this fix to next release.